core/src/test/java/org/infinispan/statetransfer/ReadAfterLostDataTest.java

```diff
@@ -217,11 +213,7 @@ private void invokeOperation(Cache<Object, Object> cache, BiFunction<Cache<Objec
       for (Object key : keys) {
          writeMap.put(key, "other" + (i++));
       }
-      ComponentRegistry cr = cache.getAdvancedCache().getComponentRegistry();
-      InvocationContext ctx = cr.getComponent(InvocationContextFactory.class).createInvocationContext(true, keys.size());
-      PutMapCommand putMapCommand = cr.getComponent(CommandsFactory.class).buildPutMapCommand(writeMap, null, 0);
-      ctx.setLockOwner(putMapCommand.getKeyLockOwner());
-      return (Map<?, ?>) cache.getAdvancedCache().getAsyncInterceptorChain().invoke(ctx, putMapCommand);
+      return cache.getAdvancedCache().getAndPutAll(writeMap);
```

Comment:
That's why the added test uses two local and two remote keys, one of which is always without mapping.